content/The_Cartridge_Header.md

```diff
@@ -208,6 +213,8 @@ Specifies the size of the external RAM in the cartridge (if any).
 When using a MBC2 chip $00 must be specified in this entry, even though
 the MBC2 includes a built-in RAM of 512 x 4 bits.
 
+Note: Ram sizes below 8KB will _repeat_ in the "cart ram" address range.
```

Comment:
Not necessarily, that depends on the mapper.

Reply:
The mapper isn't aware of the ram size outside the bank number - the address pins are wired directly to the ram chip. The chips only have as many pins as they need for their size - a 2KB ram has 11 address pins. Accessing at offset 0x0800 (2048) of the ram region will set pin 12 to 1 and the 11 to zero - only the 11 are connected, so the ram will dutifully respond with the contents of address 0, wrapping itself into the space.

In the case of MBC2, the MBC2 chip only has 9 address pins for the ram - 512 possible addresses. Higher addresses would set higher pins that aren't connected to the MBC so it just returns as if the address was masked to 9 bits, which results in it repeating, exactly the same as a separate ram chip.

Reply:
No commercial game actually used the 2KB RAM configuration so arguing that is as fruitful as arguing how ROM_RAM worked. The MBC2 is the only place this applies for commercial games (which is different anyways since it's not a separate RAM, but probably identical anyways). In fact the MBC7 would be a somewhat a counterexample of this claim (invalid address ranges of it just read out FF instead of looping back). MBCs don't necessarily have to have nothing past RAM too, they could potentially stuff it with registers of extra hardware.

Reply:
>No commercial game actually used the 2KB RAM configuration
I had a look, and it seems you're right - it's a common claim of PD roms but probably wasn't accurate if they were ever put on actual carts (most likely a flash cart with 8KB or more ram was used if any).

It seems it's only relevant for MBC2, so perhaps it would be better to put it on that page.

> In fact the MBC7 would be a somewhat a counterexample of this claim (invalid address ranges of it just read out FF instead of looping back).
MBC7 doesn't have or support RAM. The claim specifically mentions ram.

Reply:
Hence somewhat, there isn't RAM per se (the closest being the EEPROM) but what is there does not loop around as the claim would say.

Reply:
Ah but it does - the addresses are listed as like Ax5x - because they repeat every 0x100!

Reply:
> No commercial game actually used the 2KB RAM configuration

I think it might be good to mark each header option that was never seen in the wild as so.

Reply:
Anyway, I moved the note to MBC2, and added a note about the 2KB size and "PD" ROMs.